src/main/java/jsr223/perl/PerlScriptEngine.java

```diff
@@ -0,0 +1,163 @@
+
+    private PerlLog4jConfigurationLoader perlLog4JConfigurationLoader = new PerlLog4jConfigurationLoader();
+
+    public PerlScriptEngine() {
+        // This is the entry-point of the script engine
+        perlLog4JConfigurationLoader.loadLog4jConfiguration();
+    }
+
+    @Override
+    public Object eval(String script, ScriptContext context) throws ScriptException {
+
+        // Create perl command
+        String[] perlCommand = perlCommandCreator.createPerlExecutionCommand();
+
+        // Create a process builder
+        ProcessBuilder processBuilder = PerlSingletonPerlProcessBuilderFactory.getInstance()
+                                                                              .getProcessBuilder(perlCommand);
+
+        // Use process builder environment and fill it with environment variables
+        Map<String, String> variablesMap = processBuilder.environment();
+
+        // Add string bindings as environment variables
+        perlStringBindingsAdder.addBindingToStringMap(context.getBindings(ScriptContext.ENGINE_SCOPE), variablesMap);
+
+        // Replace variables in configuration file
+        String scriptReplacedVariables = perlVariablesReplacer.replaceVariables(script, variablesMap);
+
+        File perlFile = null;
+
+        try {
+            perlFile = perlConfigurationFileWriter.forceFileToDisk(scriptReplacedVariables,
+                                                                   perlCommandCreator.PERL_FILE_NAME);
+
+            // Start process
+            Process process = processBuilder.start();
+
+            // Attach streams
+            processBuilderUtilities.attachStreamsToProcess(process,
+                                                           context.getWriter(),
+                                                           context.getErrorWriter(),
+                                                           context.getReader());
+
+            // Wait for process to exit
+            int exitValue = process.waitFor();
+            if (context.getBindings(ScriptContext.ENGINE_SCOPE).containsKey(VARIABLES_BINDING_NAME)) {
+                Map<String, Serializable> variables = (Map<String, Serializable>) context.getBindings(ScriptContext.ENGINE_SCOPE)
+                                                                                         .get(VARIABLES_BINDING_NAME);
+                variables.put(EXIT_VALUE_BINDING_NAME, exitValue);
+            }
+            context.getBindings(ScriptContext.ENGINE_SCOPE).put(EXIT_VALUE_BINDING_NAME, exitValue);
+            if (exitValue != 0) {
+                throw new ScriptException("Perl process execution has failed with exit code " + exitValue);
+            }
+            return exitValue;
+        } catch (IOException e) {
+            log.warn("Failed to execute Perl.", e);
+        } catch (InterruptedException e) {
+            log.info("Container execution interrupted. " + e.getMessage());
+        } finally {
+
```

Comment:
Here you should add a process.waitFor(); 

Because when we kill a task the thread gets interrupted. Then the thread gets x amount of time to remove the process. So you should interrupt the process with process.destroy() and then let the process gracefull terminate with process.waitFor(). And then continue the execution.